src/components/Contacts/ContactFilters/ContactFilters.tsx

```diff
@@ -1,35 +1,211 @@
-import { colors } from '@material-ui/core';
-import React from 'react';
-import { useContactFiltersLazyQuery } from '../../../../pages/accountLists/[accountListId]/ContactFilters.generated';
+import {
+  Box,
+  BoxProps,
+  Button,
+  CircularProgress,
+  Collapse,
+  IconButton,
+  List,
+  ListItem,
+  ListItemText,
+  Slide,
+  styled,
+  Typography,
+  useTheme,
+} from '@material-ui/core';
+import { ArrowBackIos, ArrowForwardIos } from '@material-ui/icons';
+import { isArray } from 'lodash';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ContactFilterGroup } from '../../../../graphql/types.generated';
+import { Filter } from '../../Shared/Filters/Filter';
+import { FilterListItem } from '../../Shared/Filters/FilterListItem';
+import { FilterListItemShowAll } from '../../Shared/Filters/FilterListItemShowAll';
+import { useContactFiltersQuery } from './ContactFilters.generated';
+
+const FilterHeader = styled(Box)(({ theme }) => ({
+  padding: theme.spacing(2),
+  borderBottom: '1px solid',
+  borderBottomColor: theme.palette.grey[200],
+}));
+
+const FilterList = styled(List)(({ theme }) => ({
+  '& .MuiListItemIcon-root': {
+    minWidth: '37px',
+  },
+  '& .FilterListItemMultiselect-root': {
+    marginBottom: theme.spacing(4),
+  },
+}));
+
+const LinkButton = styled(Button)(() => ({
+  minWidth: 0,
+  textTransform: 'none',
+}));
 
 interface Props {
   accountListId: string;
 }
 
-export const ContactFilters: React.FC<Props> = ({ accountListId }: Props) => {
-  const [
-    loadContactFilters,
-    { data, loading, error },
-  ] = useContactFiltersLazyQuery({
+export const ContactFilters: React.FC<Props & BoxProps> = ({
+  accountListId,
+  ...boxProps
+}) => {
+  const theme = useTheme();
+  const { t } = useTranslation();
+
+  const { data, loading, error } = useContactFiltersQuery({
     variables: { accountListId },
   });
 
+  const [selectedGroup, showGroup] = useState<null | ContactFilterGroup>(null);
+  const [selectedFilters, setSelectedFilters] = useState({});
```

Comment:
```suggestion
  const [selectedFilters, setSelectedFilters] = useState<{
    [name: string]: unknown;
  }>({});
```

I'm not sure how to type `value`... I just stuck `unknown` here for now. It seems like some combination of string, string array, and boolean. Maybe that's good enough for now. Or maybe there's some way to set the type for each kind of filter.

I guess that can wait till we wire this up to the contacts query as that probably will have a specific format.

Reply:
yeah, I was very loose on the filter type for now due to not having an actual API defined, I'm sure some filter value cleanup will be needed once we start using this for filtering queries